public/index.css

```
@@ -78,43 +78,65 @@ nav ul {
 }
 
 .content > section > * + *,
-.content > section > blockquote > * + * {
+.content > section > blockquote > * + *,
+.content > section > :is(ul, ol) > li > * + * {
 	margin-top: 1.25rem;
+	margin-bottom: 1.25rem;
 }
 
-.content > section > * + :is(ul, ol) {
-	margin-top: 0.3rem;
+.content > section > :is(ul, ol) > li,
+.content > section > :is(ul, ol) > li > * + *,
+.content > section > :is(ul, ol) > li > :is(ul, ol) > li,
+.content > section > :is(ul, ol) > li > :is(ul, ol) > li > * + * {
+	margin-top: 0.5rem;
 }
 
 .content > section > :first-child {
 	margin-top: 0;
 }
 
+.content > section > :is(ul, ol),
+.content > section > :is(ul, ol) :is(ul, ol) {
+	padding-inline-start: 1.5em;
+}
+
 /* Typography */
-h1,
-h2,
-h3,
-h4,
-h5,
-h6 {
+:is(h1, h2, h3, h4, h5, h6),
+.content > section > .heading-wrapper {
 	margin-bottom: 1rem;
 	font-weight: bold;
 	line-height: 1.3;
 }
 
+:is(h1, h2, h3, h4, h5, h6) + *,
+.content > section > .heading-wrapper + * {
+	margin-top: 0;
+}
+
 h1,
 h2 {
 	max-width: 40ch;
 }
 
-:is(h2, h3):not(:first-child) {
+:is(h2):not(:first-child),
+.heading-wrapper:not(:first-child):is(.level-h2) {
+	margin-top: 5rem;
+}
```

Thanks @delucis, amazing work!! I think 5rem is maybe too much margin (as Sarah commented), something between 3.5 and 4 looks better to me.
```suggestion
.heading-wrapper:not(:first-child):is(.level-h2) {
	margin-top: 4rem;
}
```